rust/xaynet-mobile/src/ffi/config.rs

```diff
@@ -0,0 +1,56 @@
+use crate::ffi::{ERR_NULLPTR, OK};
+use std::os::raw::c_int;
+use xaynet_core::mask::DataType;
+
+mod pv {
+    use super::ModelConfig;
+    ffi_support::define_box_destructor!(ModelConfig, _xaynet_ffi_model_config_destroy);
+}
+
+/// Destroy the model configuration created by [`xaynet_ffi_participant_model_config()`].
+///
+/// # Return value
+///
+/// - [`OK`] on success
+/// - [`ERR_NULLPTR`] if `model_config` is NULL
+///
+/// # Safety
+///
+/// 1. When calling this method, you have to ensure that *either* the pointer is NULL
+///    *or* all of the following is true:
+///    - The pointer must be properly [aligned].
+///    - It must be "dereferencable" in the sense defined in the [`::std::ptr`] module
+///      documentation.
+/// 2. After destroying the `ModelConfig`, the pointer becomes invalid and must not be
+///    used.
+/// 3. This function should only be called on a pointer that has been created by
+///    [`xaynet_ffi_participant_model_config()`].
+///
+/// [`::std::ptr`]: https://doc.rust-lang.org/std/ptr/index.html#safety
+/// [aligned]: https://doc.rust-lang.org/std/ptr/index.html#alignment
+#[no_mangle]
+pub unsafe extern "C" fn xaynet_ffi_model_config_destroy(model_config: *mut ModelConfig) -> c_int {
+    if model_config.is_null() {
+        return ERR_NULLPTR;
+    }
+    pv::_xaynet_ffi_model_config_destroy(model_config);
+    OK
+}
+
+#[repr(C)]
+/// The model configuration of the model that is expected in [`xaynet_ffi_participant_set_model`].
+pub struct ModelConfig {
```

Comment:
maybe redundant? we could also just annotate `ModelConfig` with `#[repr(C)]` in https://github.com/xaynetwork/xaynet/pull/629/files#diff-ae8e63e3ded5ff755ba7e68cae29f44ed121b618634584aed4d1ac4b2a86a188R290

Reply:
Yeah I'm not sure to be honest. On the one hand, I think it's nice to have separate types for FFI, but on the other hand, it's a bit artificial sometimes. So I don't really have a preference